test/Sentry.Tests/HttpStatusCodeRangeExtensionsTests.cs

```diff
@@ -0,0 +1,188 @@
+namespace Sentry.Tests;
+
+public class HttpStatusCodeRangeExtensionsTests
+{
+    [Fact]
+    public void Contains_EmptyList_ReturnsFalse()
+    {
+        // Arrange
+        var ranges = new List<HttpStatusCodeRange>();
+
+        // Act
+        var result = ranges.ContainsStatusCode(404);
+
+        // Assert
+        result.Should().BeFalse();
+    }
+
+    [Theory]
+    [InlineData(400)]
+    [InlineData(450)]
+    [InlineData(499)]
+    public void Contains_SingleRange_InRange_ReturnsTrue(int statusCode)
+    {
+        // Arrange
+        var ranges = new List<HttpStatusCodeRange> { (400, 499) };
+
+        // Act
+        var result = ranges.ContainsStatusCode(statusCode);
+
+        // Assert
+        result.Should().BeTrue();
+    }
+
+    [Theory]
+    [InlineData(200)]
+    [InlineData(399)]
+    [InlineData(500)]
+    public void Contains_SingleRange_OutOfRange_ReturnsFalse(int statusCode)
+    {
+        // Arrange
+        var ranges = new List<HttpStatusCodeRange> { (400, 499) };
+
+        // Act
+        var result = ranges.ContainsStatusCode(statusCode);
+
+        // Assert
+        result.Should().BeFalse();
+    }
+
+    [Theory]
+    [InlineData(400)] // In first range
+    [InlineData(404)] // In first range
+    [InlineData(500)] // In second range
+    [InlineData(503)] // In second range
+    public void Contains_MultipleRanges_InAnyRange_ReturnsTrue(int statusCode)
+    {
+        // Arrange
+        var ranges = new List<HttpStatusCodeRange> { (400, 404), (500, 503) };
+
+        // Act
+        var result = ranges.ContainsStatusCode(statusCode);
+
+        // Assert
+        result.Should().BeTrue();
+    }
+
+    [Theory]
+    [InlineData(200)] // Below ranges
+    [InlineData(405)] // Between ranges
+    [InlineData(499)] // Between ranges
+    [InlineData(504)] // Above ranges
+    public void Contains_MultipleRanges_NotInAnyRange_ReturnsFalse(int statusCode)
+    {
+        // Arrange
+        var ranges = new List<HttpStatusCodeRange> { (400, 404), (500, 503) };
+
+        // Act
+        var result = ranges.ContainsStatusCode(statusCode);
+
+        // Assert
+        result.Should().BeFalse();
+    }
+
+    [Theory]
+    [InlineData(400)] // In first range only
+    [InlineData(425)] // In overlap
+    [InlineData(450)] // In overlap
+    [InlineData(475)] // In second range only
+    public void Contains_OverlappingRanges_InUnion_ReturnsTrue(int statusCode)
+    {
+        // Arrange
+        var ranges = new List<HttpStatusCodeRange> { (400, 450), (425, 475) };
+
+        // Act
+        var result = ranges.ContainsStatusCode(statusCode);
+
+        // Assert
+        result.Should().BeTrue();
+    }
+
+    [Theory]
+    [InlineData(200)]
+    [InlineData(399)]
+    [InlineData(476)]
+    [InlineData(500)]
```

Comment:
suggestion(nitpick): would you mind adding explanatory comments to these four tests as well? ... just as with the test above ... for consistency


```suggestion
    [InlineData(200)] // Below first range
    [InlineData(399)] // Below first range
    [InlineData(476)] // Above second range
    [InlineData(500)] // Above second range
```